### PR TITLE
Enhanced documentation for Java classes

### PR DESCRIPTION
**Pull request by javadev**

### CODE REVIEW

**Comment by javadev** [comment]
I have tested it locally

```
[INFO] >>> maven-javadoc-plugin:3.5.0:javadoc (default-cli) > generate-sources @ json >>>
[INFO] 
[INFO] <<< maven-javadoc-plugin:3.5.0:javadoc (default-cli) < generate-sources @ json <<<
[INFO] 
[INFO] 
[INFO] --- maven-javadoc-plugin:3.5.0:javadoc (default-cli) @ json ---
[INFO] No previous run data found, generating javadoc.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.844 s
[INFO] Finished at: 2024-02-02T09:07:20+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by stleary** [comment]
Looks like one of the unit tests is failing when executed with Java 11. This will be fixed in a separate PR.

File: pom.xml [hunk 1]
```diff
@@ -126,6 +126,9 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.5.0</version>
+                <configuration>
```
Comment (stleary) [inline pom.xml:129]: Please remove, since this does not seem necessary for fixing the docs.

Comment (javadev) [reply pom.xml:129]: I came across a Javadoc warning without having this configuration.

**Comment by stleary** [comment]
**What problem does this code solve?**
Addresses compiler warnings by adding JavaDoc comments to some methods

**Does the code still compile with Java6?**
Yes, but there is a pom.xml change that specifies Java 8, to fix a compiler warning.
This will be reverted in a follow-on commit.

**Risks**
Low

**Changes to the API?**
No

**Will this require a new release?**
No

**Should the documentation be updated?**
No

**Does it break the unit tests?**
No

**Was any code refactored in this commit?**
No

**Review status**
**APPROVED**

Starting 3-day comment window

**Review by stleary — APPROVED** [review]

**Pull request merged**